### ISSUE TITLE
Cannot reproduce Panoptic DeepLab on COCO

### ISSUE THREAD

**Issue by jd730** [facebookresearch/detectron2#2934]
Hi, @bowenc0221

I ran Panoptic DeepLab with batch size 64 (16 * 4 GPUs) with default hyperparameters, 200000 iteration, base_lr: 0.0005. (i.e., I change IMS_PER_BATCH from 64 to 16).

However, I found that due to https://github.com/facebookresearch/detectron2/blob/a68f7ea6a9335a54762bfecf7b8f0a186bab8ed8/detectron2/config/defaults.py#L525, learning rate is decayed at 30000 iteration.

I found that the performance is much lower than expected. 

![image](https://user-images.githubusercontent.com/9247850/115320369-b8209a00-a1bc-11eb-9c53-1b1eeeae09c6.png)

Unfortunately, when I manually change STEPS to 300000 (just a large number) following the original paper, the performance is still low. 

Did I miss some details?

Thank you!

1. Full runnable code or full changes you made:
Change IMS_PER_BATCH from 64 to 16 in https://github.com/facebookresearch/detectron2/blob/a68f7ea6a9335a54762bfecf7b8f0a186bab8ed8/projects/Panoptic-DeepLab/configs/COCO-PanopticSegmentation/panoptic_deeplab_R_52_os16_mg124_poly_200k_bs64_crop_640_640_coco_dsconv.yaml#L30

2. What exact command you run:
 ```
CUDA_VISIBLE_DEVICES=7,6,5,4 python train_net.py --config configs/COCO-PanopticSegmentation/panoptic_deeplab_R_52_os16_mg124_poly_200k_bs64_crop_640_640_coco_dsconv.yaml  --num-gpus 4
```
3. __Full logs__ or other relevant observations:
I attached full log.
[log.txt](https://github.com/facebookresearch/detectron2/files/6339894/log.txt)

## Expected behavior:

If there are no obvious crash in "full logs" provided above,
please tell us the expected behavior.

**Comment by ppwwyyxx** [comment]
> , learning rate is decayed at 30000 iteration.

`STEPS` is not used by this config and learning rate will not decay at 30000 iteration.

> Change IMS_PER_BATCH from 64 to 16 in

Because config is changed, it is expected that the results are not reproduced.
You might need to also change other configs (learning rate, training length, etc) to get better performance.

**Comment by jd730** [comment]
Hi @ppwwyyxx,
From my understanding, due to https://github.com/facebookresearch/detectron2/blob/02d9c3f43402d8746aa77f9850f27ecb9a658f08/projects/Panoptic-DeepLab/train_net.py#L140, the model call default parameters and overwrite configs in https://github.com/facebookresearch/detectron2/blob/a68f7ea6a9335a54762bfecf7b8f0a186bab8ed8/projects/Panoptic-DeepLab/configs/COCO-PanopticSegmentation/panoptic_deeplab_R_52_os16_mg124_poly_200k_bs64_crop_640_640_coco_dsconv.yaml.

Here is actual configuration that is printed before training (see previously attached file) when I ran using https://github.com/facebookresearch/detectron2/blob/a68f7ea6a9335a54762bfecf7b8f0a186bab8ed8/projects/Panoptic-DeepLab/configs/COCO-PanopticSegmentation/panoptic_deeplab_R_52_os16_mg124_poly_200k_bs64_crop_640_640_coco_dsconv.yaml.


```
_BASE_: ../Cityscapes-PanopticSegmentation/Base-PanopticDeepLab-OS16.yaml
MODEL:
  WEIGHTS: "detectron2://DeepLab/R-52.pkl"
  PIXEL_MEAN: [123.675, 116.280, 103.530]
  PIXEL_STD: [58.395, 57.120, 57.375]
  BACKBONE:
    NAME: "build_resnet_deeplab_backbone"
  RESNETS:
    DEPTH: 50
    NORM: "SyncBN"
    RES5_MULTI_GRID: [1, 2, 4]
    STEM_TYPE: "deeplab"
    STEM_OUT_CHANNELS: 128
    STRIDE_IN_1X1: False
  SEM_SEG_HEAD:
    NUM_CLASSES: 133
    LOSS_TOP_K: 1.0
    USE_DEPTHWISE_SEPARABLE_CONV: True
  PANOPTIC_DEEPLAB:
    STUFF_AREA: 4096
    NMS_KERNEL: 41
    SIZE_DIVISIBILITY: 640
    USE_DEPTHWISE_SEPARABLE_CONV: True
DATASETS:
  TRAIN: ("coco_2017_train_panoptic",)
  TEST: ("coco_2017_val_panoptic",)
SOLVER:
  BASE_LR: 0.0005
  MAX_ITER: 200000
  IMS_PER_BATCH: 8
INPUT:
  FORMAT: "RGB"
  GAUSSIAN_SIGMA: 8
  MIN_SIZE_TRAIN: !!python/object/apply:eval ["[int(x * 0.1 * 640) for x in range(5, 16)]"]
  MIN_SIZE_TRAIN_SAMPLING: "choice"
  MIN_SIZE_TEST: 640
  MAX_SIZE_TRAIN: 960
  MAX_SIZE_TEST: 640
  CROP:
    ENABLED: True
    TYPE: "absolute"
    SIZE: (640, 640)
OUTPUT_DIR: jdhwang/re_vanila_panoptic_deeplab # results/panoptic_deeplab_bs128

[04/18 12:48:35] detectron2 INFO: Running with full config:
CUDNN_BENCHMARK: False
DATALOADER:
  ASPECT_RATIO_GROUPING: True
  FILTER_EMPTY_ANNOTATIONS: True
  NUM_WORKERS: 10
  REPEAT_THRESHOLD: 0.0
  SAMPLER_TRAIN: TrainingSampler
DATASETS:
  PRECOMPUTED_PROPOSAL_TOPK_TEST: 1000
  PRECOMPUTED_PROPOSAL_TOPK_TRAIN: 2000
  PROPOSAL_FILES_TEST: ()
  PROPOSAL_FILES_TRAIN: ()
  TEST: ('coco_2017_val_panoptic',)
  TRAIN: ('coco_2017_train_panoptic',)
GLOBAL:
  HACK: 1.0
INPUT:
  CROP:
    ENABLED: True
    SINGLE_CATEGORY_MAX_AREA: 1.0
    SIZE: [640, 640]
    TYPE: absolute
  FORMAT: RGB
  GAUSSIAN_SIGMA: 8
  IGNORE_CROWD_IN_SEMANTIC: False
  IGNORE_STUFF_IN_OFFSET: True
  MASK_FORMAT: polygon
  MAX_SIZE_TEST: 640
  MAX_SIZE_TRAIN: 960
  MIN_SIZE_TEST: 640
  MIN_SIZE_TRAIN: (320, 384, 448, 512, 576, 640, 704, 768, 832, 896, 960)
  MIN_SIZE_TRAIN_SAMPLING: choice
  RANDOM_FLIP: horizontal
  SMALL_INSTANCE_AREA: 4096
  SMALL_INSTANCE_WEIGHT: 3
MODEL:
  ANCHOR_GENERATOR:
    ANGLES: [[-90, 0, 90]]
    ASPECT_RATIOS: [[0.5, 1.0, 2.0]]
    NAME: DefaultAnchorGenerator
    OFFSET: 0.0
    SIZES: [[32, 64, 128, 256, 512]]
  BACKBONE:
    FREEZE_AT: 0
    NAME: build_resnet_deeplab_backbone
  DEVICE: cuda
  FPN:
    FUSE_TYPE: sum
    IN_FEATURES: []
    NORM: 
    OUT_CHANNELS: 256
  INS_EMBED_HEAD:
    ASPP_CHANNELS: 256
    ASPP_DILATIONS: [6, 12, 18]
    ASPP_DROPOUT: 0.1
    CENTER_LOSS_WEIGHT: 200.0
    COMMON_STRIDE: 4
    CONVS_DIM: 128
    HEAD_CHANNELS: 32
    IN_FEATURES: ['res2', 'res3', 'res5']
    NAME: PanopticDeepLabInsEmbedHead
    NORM: SyncBN
    OFFSET_LOSS_WEIGHT: 0.01
    PROJECT_CHANNELS: [32, 64]
    PROJECT_FEATURES: ['res2', 'res3']
  KEYPOINT_ON: False
  LOAD_PROPOSALS: False
  MASK_ON: False
  META_ARCHITECTURE: PanopticDeepLab
  PANOPTIC_DEEPLAB:
    BENCHMARK_NETWORK_SPEED: False
    CENTER_THRESHOLD: 0.1
    NMS_KERNEL: 41
    PREDICT_INSTANCES: True
    SIZE_DIVISIBILITY: 640
    STUFF_AREA: 4096
    TOP_K_INSTANCE: 200
    USE_DEPTHWISE_SEPARABLE_CONV: True
  PANOPTIC_FPN:
    COMBINE:
      ENABLED: True
      INSTANCES_CONFIDENCE_THRESH: 0.5
      OVERLAP_THRESH: 0.5
      STUFF_AREA_LIMIT: 4096
    INSTANCE_LOSS_WEIGHT: 1.0
  PIXEL_MEAN: [123.675, 116.28, 103.53]
  PIXEL_STD: [58.395, 57.12, 57.375]
  PROPOSAL_GENERATOR:
    MIN_SIZE: 0
    NAME: RPN
  RESNETS:
    DEFORM_MODULATED: False
    DEFORM_NUM_GROUPS: 1
    DEFORM_ON_PER_STAGE: [False, False, False, False]
    DEPTH: 50
    NORM: SyncBN
    NUM_GROUPS: 1
    OUT_FEATURES: ['res2', 'res3', 'res5']
    RES2_OUT_CHANNELS: 256
    RES4_DILATION: 1
    RES5_DILATION: 2
    RES5_MULTI_GRID: [1, 2, 4]
    STEM_OUT_CHANNELS: 128
    STEM_TYPE: deeplab
    STRIDE_IN_1X1: False
    WIDTH_PER_GROUP: 64
  RETINANET:
    BBOX_REG_LOSS_TYPE: smooth_l1
    BBOX_REG_WEIGHTS: (1.0, 1.0, 1.0, 1.0)
    FOCAL_LOSS_ALPHA: 0.25
    FOCAL_LOSS_GAMMA: 2.0
    IN_FEATURES: ['p3', 'p4', 'p5', 'p6', 'p7']
    IOU_LABELS: [0, -1, 1]
    IOU_THRESHOLDS: [0.4, 0.5]
    NMS_THRESH_TEST: 0.5
    NORM: 
    NUM_CLASSES: 80
    NUM_CONVS: 4
    PRIOR_PROB: 0.01
    SCORE_THRESH_TEST: 0.05
    SMOOTH_L1_LOSS_BETA: 0.1
    TOPK_CANDIDATES_TEST: 1000
  ROI_BOX_CASCADE_HEAD:
    BBOX_REG_WEIGHTS: ((10.0, 10.0, 5.0, 5.0), (20.0, 20.0, 10.0, 10.0), (30.0, 30.0, 15.0, 15.0))
    IOUS: (0.5, 0.6, 0.7)
  ROI_BOX_HEAD:
    BBOX_REG_LOSS_TYPE: smooth_l1
    BBOX_REG_LOSS_WEIGHT: 1.0
    BBOX_REG_WEIGHTS: (10.0, 10.0, 5.0, 5.0)
    CLS_AGNOSTIC_BBOX_REG: False
    CONV_DIM: 256
    FC_DIM: 1024
    NAME: 
    NORM: 
    NUM_CONV: 0
    NUM_FC: 0
    POOLER_RESOLUTION: 14
    POOLER_SAMPLING_RATIO: 0
    POOLER_TYPE: ROIAlignV2
    SMOOTH_L1_BETA: 0.0
    TRAIN_ON_PRED_BOXES: False
  ROI_HEADS:
    BATCH_SIZE_PER_IMAGE: 512
    IN_FEATURES: ['res4']
    IOU_LABELS: [0, 1]
    IOU_THRESHOLDS: [0.5]
    NAME: Res5ROIHeads
    NMS_THRESH_TEST: 0.5
    NUM_CLASSES: 80
    POSITIVE_FRACTION: 0.25
    PROPOSAL_APPEND_GT: True
    SCORE_THRESH_TEST: 0.05
  ROI_KEYPOINT_HEAD:
    CONV_DIMS: (512, 512, 512, 512, 512, 512, 512, 512)
    LOSS_WEIGHT: 1.0
    MIN_KEYPOINTS_PER_IMAGE: 1
    NAME: KRCNNConvDeconvUpsampleHead
    NORMALIZE_LOSS_BY_VISIBLE_KEYPOINTS: True
    NUM_KEYPOINTS: 17
    POOLER_RESOLUTION: 14
    POOLER_SAMPLING_RATIO: 0
    POOLER_TYPE: ROIAlignV2
  ROI_MASK_HEAD:
    CLS_AGNOSTIC_MASK: False
    CONV_DIM: 256
    NAME: MaskRCNNConvUpsampleHead
    NORM: 
    NUM_CONV: 0
    POOLER_RESOLUTION: 14
    POOLER_SAMPLING_RATIO: 0
    POOLER_TYPE: ROIAlignV2
  RPN:
    BATCH_SIZE_PER_IMAGE: 256
    BBOX_REG_LOSS_TYPE: smooth_l1
    BBOX_REG_LOSS_WEIGHT: 1.0
    BBOX_REG_WEIGHTS: (1.0, 1.0, 1.0, 1.0)
    BOUNDARY_THRESH: -1
    CONV_DIMS: [-1]
    HEAD_NAME: StandardRPNHead
    IN_FEATURES: ['res4']
    IOU_LABELS: [0, -1, 1]
    IOU_THRESHOLDS: [0.3, 0.7]
    LOSS_WEIGHT: 1.0
    NMS_THRESH: 0.7
    POSITIVE_FRACTION: 0.5
    POST_NMS_TOPK_TEST: 1000
    POST_NMS_TOPK_TRAIN: 2000
    PRE_NMS_TOPK_TEST: 6000
    PRE_NMS_TOPK_TRAIN: 12000
    SMOOTH_L1_BETA: 0.0
  SEM_SEG_HEAD:
    ASPP_CHANNELS: 256
    ASPP_DILATIONS: [6, 12, 18]
    ASPP_DROPOUT: 0.1
    COMMON_STRIDE: 4
    CONVS_DIM: 256
    HEAD_CHANNELS: 256
    IGNORE_VALUE: 255
    IN_FEATURES: ['res2', 'res3', 'res5']
    LOSS_TOP_K: 1.0
    LOSS_TYPE: hard_pixel_mining
    LOSS_WEIGHT: 1.0
    NAME: PanopticDeepLabSemSegHead
    NORM: SyncBN
    NUM_CLASSES: 133
    PROJECT_CHANNELS: [32, 64]
    PROJECT_FEATURES: ['res2', 'res3']
    USE_DEPTHWISE_SEPARABLE_CONV: True
  WEIGHTS: detectron2://DeepLab/R-52.pkl
OPEN:
  DEPTH: False
  UNKNOWN_PATH: 
  VOID: False
OUTPUT_DIR: jdhwang/re_vanila_panoptic_deeplab
SEED: -1
SOLVER:
  AMP:
    ENABLED: False
  BASE_LR: 0.0005
  BIAS_LR_FACTOR: 1.0
  CHECKPOINT_PERIOD: 5000
  CLIP_GRADIENTS:
    CLIP_TYPE: value
    CLIP_VALUE: 1.0
    ENABLED: False
    NORM_TYPE: 2.0
  GAMMA: 0.1
  IMS_PER_BATCH: 8
  LR_SCHEDULER_NAME: WarmupPolyLR
  MAX_ITER: 200000
  MOMENTUM: 0.9
  NESTEROV: False
  OPTIMIZER: ADAM
  POLY_LR_CONSTANT_ENDING: 0.0
  POLY_LR_POWER: 0.9
  REFERENCE_WORLD_SIZE: 0
  STEPS: (30000,)
  WARMUP_FACTOR: 0.001
  WARMUP_ITERS: 1000
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0
  WEIGHT_DECAY_BIAS: 0.0
  WEIGHT_DECAY_NORM: 0.0
TEST:
  AUG:
    ENABLED: False
    FLIP: True
    MAX_SIZE: 4000
    MIN_SIZES: (400, 500, 600, 700, 800, 900, 1000, 1100, 1200)
  DETECTIONS_PER_IMAGE: 100
  EVAL_PERIOD: 0
  EXPECTED_RESULTS: []
  KEYPOINT_OKS_SIGMAS: []
  PRECISE_BN:
    ENABLED: False
    NUM_ITER: 200
VERSION: 2
VIS_PERIOD: 0
```

Moreover, if I understand correctly, IMS_PER_BATCH denotes batch size per each gpus. So, if I set a different number of GPUS, I need to change it.

According to the original paper, the batch size for COCO is 64 and I used 4 GPUS. Isn't it correct that I need to change IMS_PER_BATCH to 16 so that the total batch size is 64?

Thank you!

![image](https://user-images.githubusercontent.com/9247850/115483014-e7054180-a28a-11eb-9bea-844a13b56f3d.png)

**Comment by bowenc0221** [comment]
>, learning rate is decayed at 30000 iteration.

Panoptic-DeepLab uses the `Poly` learning rate scheduler, so STEPS is not used.

>Change IMS_PER_BATCH from 64 to 16 in

IMS_PER_BATCH is the total number of images for all GPUs

**Comment by jd730** [comment]
Thank @bowenc0221, @ppwwyyxx! I will try again with IMS_PER_BATCH 64.

**Comment by jd730** [comment]
Hi @bowenc0221  @ppwwyyxx ,
I ran with the default setting (i.e., IMS_PER_BATCH: 64 and I did not change STEPS) as you recommended but the result is still much lower than the paper. 

![image](https://user-images.githubusercontent.com/9247850/115939260-8bc88e80-a4d8-11eb-99a8-80dc43581bea.png)

Did I miss something?

Here is the [log](https://github.com/facebookresearch/detectron2/files/6368549/log.txt) and [metrics](https://github.com/facebookresearch/detectron2/files/6368555/metrics.txt)

Thank you so much.

**Comment by ppwwyyxx** [comment]
With no changes to any code on github, I trained with the config two days ago and was able to reproduce the performance in README.
